### PR TITLE
[EngLife] [KH-28794] Undo Initial Formatting Changes

### PR DESCRIPTION
**Pull request by isaisarin**
[KH-28794](https://kohoteam.atlassian.net/browse/KH-28794)

This reverts the formatting changes done in commit e2946ae3a7736210a5fea4f873cdba6be7d5ed46. This will allow different format strings to retrieve different amounts of the error (importantly with or without stack trace).

Since this is a big change, this PR also fixes all tests and adds a workflow to run them.

Relevant discussion: https://github.com/kohofinancial/errors/discussions/15

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Go
```
Comment (bewuethr) [inline .github/workflows/go.yml:1]:
```suggestion
name: Run Go tests
```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+on: [push]
```
Comment (bewuethr) [inline .github/workflows/go.yml:3]:
```suggestion
on:
  - push
```

[Reference](https://github.com/kohofinancial/documentation/blob/main/best%20practices/github-actions.md#cosmetics)

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  Test:
+    runs-on: ubuntu-latest
```
Comment (bewuethr) [inline .github/workflows/go.yml:7]:
```suggestion
    runs-on: ubuntu-20.04
```

[Reference](https://github.com/kohofinancial/documentation/blob/main/best%20practices/github-actions.md#version-pinning-and-dependency-management)

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (bewuethr) [inline .github/workflows/go.yml:10]:
```suggestion
      - uses: actions/checkout@v3.0.2
```

[Reference](https://github.com/kohofinancial/documentation/blob/main/best%20practices/github-actions.md#version-pinning-and-dependency-management)

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          path: github.com/kohofinancial/errors
```
Comment (bewuethr) [inline .github/workflows/go.yml:12]: Can't we just drop this? The path for the stack trace should be determined by the module path.

File: .github/workflows/go.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
```
Comment (bewuethr) [inline .github/workflows/go.yml:17]: This is an ancient version, and you can tell setup-go to use the Go version from the `go.mod` file:

```suggestion
        uses: actions/setup-go@v3.3.0
        with:
          go-version-file: go.mod
```

That's the path to the `go.mod` file (see [docs](https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file)), i.e., would have to change if you set a path in the checkout step.

File: .github/workflows/go.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+      # On this stage, we need the path to be the Github repository path for proper stack traces.
+      - name: Test
+        run: |
+          cd github.com/kohofinancial/errors
+          go test -v ./...
```
Comment (bewuethr) [inline .github/workflows/go.yml:23]: If you need the path (I don't think you do), you could probably shorten to

```suggestion
          go test -v github.com/kohofinancial/errors/...
```

**Review by bewuethr — COMMENTED** [review]
Can you also add a dependabot config file so the actions get updates? Similar to, e.g., [this one](https://github.com/kohofinancial/actions/blob/main/.github/dependabot.yml). I'm also okay with adding Nomads as codeowners to this, and/or yourself if you want to.

File: .github/workflows/go.yml (hunk 5 above)
Comment (isaisarin) [reply .github/workflows/go.yml:12]: With the local tests is uses the local path like [this run](https://github.com/kohofinancial/errors/actions/runs/3084773541/jobs/4987308517) which I also believe is expected based on the use of regex in `want`:

```
    stack_test.go:166: test 4: line 2: fmt.Sprintf("%+v", err):
         got: "github.com/kohofinancial/errors.TestStackTrace\n\t/home/runner/work/errors/errors/stack_test.go:136"
        want: "github.com/kohofinancial/errors.TestStackTrace\n\t.+/github.com/kohofinancial/errors/stack_test.go:136"
```

Comment (bewuethr) [reply .github/workflows/go.yml:12]: Ah, okay, ignore then

File: .github/workflows/go.yml (hunk 1 above)
Comment (isaisarin) [reply .github/workflows/go.yml:1]: :thinking: This one didn't show up on the files page for the batch.

File: .github/workflows/go.yml (hunk 7 above)
Comment (isaisarin) [reply .github/workflows/go.yml]: Opted to add a default path for run instead. Didn't seem to be a way to do it globally for all steps though.

**Review by bewuethr — APPROVED** [review]

**Comment by isaisarin** [comment]
@bewuethr when merged, I'm also thinking of releasing as v2.0.0 to make upgrades which will change error logging intentional. Does this make sense?

**Review by NikitaBaranov-Koho — APPROVED** [review]
I believe v2 definitely make sense

**Comment by bewuethr** [comment]
No, don't upgrade the major version, please. We have always asked people to use `%+v` to get a stack trace, and it's used like that in most places; if we change the import path by making a major release, we have to change almost every single package in our entire ecosystem. There shouldn't be surprising changes in behaviour if we release without a major bump.

Major release would probably mean [changing the default branch here](https://go.dev/doc/modules/major-version), or [start having a `v2` directory](https://go.dev/blog/v2-go-modules), both of which feel like a huge pain and deviation from the behaviour in every other repo.

**Comment by isaisarin** [comment]
Makes sense, will do v1.4.0 then. I'll create the release shortly when I also have time to be the first person to implement it in a repository & will post about the change in backend_dev.

**Comment by bewuethr** [comment]
Remember that some repos will get dependabot PRs to update as early as the next day; maybe do a [pre-release](https://go.dev/doc/modules/version-numbers#pre-release-version) like `v1.4.0-beta` first? I believe dependabot ignores those.

**Pull request merged**